x-pack/plugins/monitoring/server/lib/setup/collection/get_collection_status.js

```diff
@@ -0,0 +1,365 @@
+  }
+
+  if (get(logstashDetectionResponse1, 'hits.total.value', 0) > 0 || get(logstashDetectionResponse2, 'hits.total.value', 0) > 0) {
+    result[LOGSTASH_SYSTEM_ID].mightExist = true;
+  }
+
+  if (get(apmDetectionResponse, 'hits.total.value', 0) > 0) {
+    result[APM_CUSTOM_ID].mightExist = true;
+  }
+
+  return result;
+}
+
+function getUuidBucketName(productName) {
+  switch (productName) {
+    case 'elasticsearch':
+      return 'es_uuids';
+    case KIBANA_SYSTEM_ID:
+      return 'kibana_uuids';
+    case BEATS_SYSTEM_ID:
+    case APM_CUSTOM_ID:
+      return 'beats_uuids';
+    case LOGSTASH_SYSTEM_ID:
+      return 'logstash_uuids';
+  }
+}
+
+function isBeatFromAPM(bucket) {
+  if (!bucket.beat_type) {
+    return false;
+  }
+
+  return get(bucket, 'beat_type.buckets[0].key') === 'apm-server';
+}
+
+function shouldSkipBucket(product, bucket) {
+  if (product.name === BEATS_SYSTEM_ID && isBeatFromAPM(bucket)) {
+    return true;
+  }
+  if (product.name === APM_CUSTOM_ID && !isBeatFromAPM(bucket)) {
+    return true;
+  }
+  return false;
+}
+
+/**
+ * This function will scan all monitoring documents within the past 30s (or a custom time range is supported too)
+ * and determine which products fall into one of three states:
+ * - isPartiallyMigrated: This means we are seeing some monitoring documents from MB and some from internal collection
+ * - isFullyMigrated: This means we are only seeing monitoring documents from MB
+ * - isInternalCollector: This means we are only seeing monitoring documents from internal collection
+ *
+ * If a product is partially migrated, this function will also return the timestamp of the last seen monitoring
+ * document from internal collection. This will help the user understand if they successfully disabled internal
+ * collection and just need to wait for the time window of the query to exclude the older, internally collected documents
+ *
+ * If a product is not detected at all (no monitoring documents), we will attempt to do some self discovery
+ * based on assumptions around indices that might exist with various products. We will return something
+ * like this in that case:
+ * detected: {
+ *   doesExist: boolean, // This product definitely exists but does not have any monitoring documents (kibana and ES)
+ *   mightExist: boolean, // This product might exist based on the presence of some other indices
+ * }
+
+ * @param {*} req Standard request object. Can contain a timeRange to use for the query
+ * @param {*} indexPatterns Map of index patterns to search against (will be all .monitoring-* indices)
+ * @param {*} clusterUuid Optional and will be used to filter down the query if used
+ */
+export const getCollectionStatus = async (req, indexPatterns, clusterUuid) => {
+  const PRODUCTS = [
+    { name: KIBANA_SYSTEM_ID },
+    { name: BEATS_SYSTEM_ID },
+    { name: LOGSTASH_SYSTEM_ID },
+    { name: APM_CUSTOM_ID, token: '-beats-' },
+    { name: 'elasticsearch', token: '-es-' },
```

Comment:
Now that I see the `elasticsearch` literal here, I'd definitely consider extracting this into a constant along side `APM_CUSTOM_ID`, per my earlier comment: https://github.com/elastic/kibana/pull/34871/files#r275830829.

Reply:
Still need to replace this literal?

Reply:
Yes, thanks